types/redux-first-router/index.d.ts

```diff
@@ -152,9 +183,15 @@ export interface Options {
     onAfterChange?<S>(dispatch: Dispatch<S>, getState: StateGetter): void;
     onBackNext?<S>(dispatch: Dispatch<S>, getState: StateGetter): void;
     restoreScroll?(history: History): ScrollBehavior;
+    initialDispatch?: boolean;
+    querySerializer?: QuerySerializer;
     navigators?: NavigatorsConfig;
 }
 
+// Params and Payload differ from Flow definition, as they can be arbitrarily shaped
+export type Params = any;
+export type Payload = any;
+
```

Comment:
@sheetalkamat you're right on all accounts. My issue was in coercing `LocationState`'s payload to match what I expected, but I can do that on my end without mangling the typings. I'll revert the types back to object, sorry for the trouble to you and @Valbrand